sdk/nodejs/npm/workspaces.go

```diff
@@ -48,11 +48,11 @@ func FindWorkspaceRoot(programDirectory string) (string, error) {
 		}
 		for _, workspace := range workspaces {
 			// See if any of the workspace glob results is the programDirectory.
-			paths, err := filepath.Glob(filepath.Join(currentDir, workspace))
+			paths, err := filepath.Glob(filepath.Join(currentDir, workspace, "package.json"))
```

Comment:
By including `package.json` in the pattern we limit the number of paths returned to those we're really interested in.